### ISSUE TITLE
SQL Syntax Errors in squawk-cli 2.9.0 for Valid PostgreSQL Statements

### ISSUE THREAD

**Issue by hc-12** [sbdchd/squawk#509]
## Summary
squawk-cli (version 2.9.0) incorrectly reports syntax errors for valid PostgreSQL SQL statements, particularly those containing Common Table Expressions (CTEs), window functions, and subqueries with aliases.

## Example Code
```sql
-- This works in PostgreSQL but fails squawk's parser
WITH ranked_notifications AS (
  SELECT 
    notification_id,
    ROW_NUMBER() OVER (
      PARTITION BY user_id, board_id ORDER BY created_at DESC
    ) AS rn
  FROM public.notification
  WHERE android_channel_id = 'watchlist'
)
UPDATE public.notification
SET dismissed_at = current_timestamp
WHERE notification_id IN (
  SELECT notification_id FROM ranked_notifications WHERE rn > 1
);
```

## Actual Behavior
Running squawk against this SQL generates multiple syntax errors such as:
```
error[syntax-error]: expected R_PAREN
error[syntax-error]: expected DELETE, SELECT, TABLE, UPDATE, or MERGE, got: AS_KW
error[syntax-error]: expected command, found AS_KW
error[syntax-error]: expected command, found FROM_KW
error[syntax-error]: expected command, found IDENT
error[syntax-error]: expected command, found WHERE_KW
```

## Expected Behavior
The linter should correctly parse standard PostgreSQL syntax and not report syntax errors for valid SQL code that executes successfully in PostgreSQL databases.

## Observations
1. **Problem constructs**: The parser particularly struggles with:
   - Common Table Expressions (WITH clauses)
   - Window functions (OVER, PARTITION BY, etc.)
   - Subqueries with aliases (using AS keyword)
   - Schema-qualified table names (e.g., public.notification)
   
2. **Cascading errors**: A single error often leads to multiple subsequent syntax errors as the parser fails to recover.

3. **Context**: These SQL statements execute successfully in PostgreSQL databases but cannot be linted without errors by squawk.

## Environment Information
- squawk-cli version: 2.9.0
- OS: macOS

**Comment by sbdchd** [comment]
Fixed in v2.10.0!

```
❯ npx squawk-cli --version
squawk 1.6.0
```

```
❯ npx squawk-cli <<EOF
-- This works in PostgreSQL but fails squawk's parser
WITH ranked_notifications AS (
  SELECT
    notification_id,
    ROW_NUMBER() OVER (
      PARTITION BY user_id, board_id ORDER BY created_at DESC
    ) AS rn
  FROM public.notification
  WHERE android_channel_id = 'watchlist'
)
UPDATE public.notification
SET dismissed_at = current_timestamp
WHERE notification_id IN (
  SELECT notification_id FROM ranked_notifications WHERE rn > 1
);
EOF
Found 0 issues in 1 file 🎉
```

**Comment by hc-12** [comment]
@sbdchd thank you for the prompt fix
